### PR TITLE
DefaultDocumentWorker customizable via envvar

### LINKED ISSUE #76: Ability to customise DefaultDocumentWorker
Is it possible to customise the value for `DefaultDocumentWorker` on linux?

By default it appears to be hard coded to `/usr/bin`:
https://github.com/aws/amazon-ssm-agent/blob/e092980b261c432156f738dc1531666bf8277679/agent/appconfig/constants_unix.go#L79 

We're trying to deploy on Container Linux based on [this blog post](https://medium.com/levops/how-to-work-with-aws-simple-system-manager-on-coreos-4741853dfd50). It's not a current supported OS, although the agent does work fine if we rebuild after changing that value. The reason it needs to be different is on Container Linux `/usr` is read only to allow for OS updates.

### PR DESCRIPTION
**Pull request by mumoshu**
Resolves #76

### CODE REVIEW

File: agent/appconfig/constants_unix.go [hunk 1]
```diff
@@ -111,5 +111,9 @@ func init() {
 			DefaultDocumentWorker = "/snap/amazon-ssm-agent/current/ssm-document-worker"
 
 		}
+		ssmDefaultDocumentWorkerFromEnv := os.Getenv("AWS_SSM_DEFAULT_DOCUMENT_WORKER")
```
Comment (mumoshu) [inline agent/appconfig/constants_unix.go]: I tried to make the envvar look official-like as much as possible.
Do you like it? 🤔

File: agent/appconfig/constants_unix.go [hunk 2]
```diff
@@ -111,5 +111,9 @@ func init() {
...
+		if ssmDefaultDocumentWorkerFromEnv != "" {
```
Comment (mumoshu) [inline agent/appconfig/constants_unix.go]: So this affects nothing if you don't explicitly set the envvar.

**Comment by mumoshu** [comment]
The travis build is failing but it has nothing to do with this PR!

See other irrelevant PRs to verity that they are also failing due to the same error.

**Comment by thedevopsmachine** [comment]
This error is due to a bug in the s3util package. I tried to fix it but the PR was never accepted because the Travis build still broke in another test (also unrelated to my fix). 

Essentially the bug is caused because the tests are running in a container, and SSM isn't supported in a container environment. It's neither an EC2 instance, nor is it an on-prem managed instance. In either one of those scenarios, the code would never fail, but it's not, so yeah, don't know what to do about that.

Here's the fix:
https://github.com/aws/amazon-ssm-agent/pull/97/commits/77dd587582beef6feacede78c0e46ccb680bbfbe

**Comment by mumoshu** [comment]
@thedevopsmachine Thanks for sharing it. I'll have a deeper look!

**Comment by mumoshu** [comment]
Also, I hope one of maintainers chimes in and provide us a direction 😉 
Can we merge this anyway and fix the build in another issues? Feel free to assign me if you like.

**Comment by mumoshu** [comment]
@nehalaws Hey! Which PR fixed this?

**Comment by nehalaws** [comment]
This PR was closed incorrectly because of a release, opening it again. Please rebase your branch with master.

**Comment by mumoshu** [comment]
@nehalaws Thanks for the response! Rebased.

**Comment by felipecoe** [comment]
I would greatly benefit from this. Can this be merged already?

**Comment by hiimtu** [comment]
Hi @mumoshu I will get the team to look at this PR. Our team is currently planning on completely moving to github for development so it will be easier in the future for customers like you to contribute. Sorry for the delay and thanks for the contribution!

**Comment by hiimtu** [comment]
https://github.com/aws/amazon-ssm-agent/commit/bc7e78017a7802854fb9ddf313ca053a20e6a471

**Pull request closed without merging**